### PR TITLE
Add all changes since last release to Change Log

### PR DESCRIPTION
**Pull request by handrews**
As promised in Monday's call, here is the change log i had researched covering all changes to date (plus PR #1359 which was agreed to in principle on the call as well).

### CODE REVIEW

**Review by jdesrosiers — COMMENTED** [review]
That's impressively thorough! I don't think clarifications need a change log entry. I'd just include things that effect behavior.

File: jsonschema-core.xml [hunk 1]
```diff
@@ -4295,10 +4295,34 @@ https://example.com/schemas/common#/$defs/allOf/1
...
                     <t hangText="draft-bhutton-json-schema-next">
                         <list style="symbols">
-                            <t>"contains" now applies to objects as well as arrays</t>
-                            <t>Use IRIs instead of URIs</t>
+                            <t>Use IRIs instead of URIs, including allowing unicode in plain-name fragments</t>
+                            <t>Clarify that detecting duplicate IRIs for different schemas SHOULD raise an error</t>
+                            <t>Consolidate and clarify the syntax and rationale for plain-name fragments</t>
+                            <t>"$id" MUST be an absolute-IRI, without any fragment, even an empty one</t>
+                            <t>Note that an empty string "$id" results in duplicate IRIs for different schemas</t>
+                            <t>Define empty schemas as empty (no longer allowing unrecognized keywords</t>
```
Comment (Julian) [inline jsonschema-core.xml:4303]:
```suggestion
                            <t>Define empty schemas as empty (no longer allowing unrecognized keywords)</t>
```

**Comment by Julian** [comment]
> That's impressively thorough! I don't think clarifications need a change log entry. I'd just include things that effect behavior.

I agree though another riff on perhaps the same underlying suggestion is attempting to sort the list by some imprecise notion of "likelihood someone cares to read about it", which should be doable. E.g. `contains` changes are probably top of the list, and all the clarifying stuff likely falls near the bottom, if it isn't in fact cut out entirely.

**Comment by gregsdennis** [comment]
Replaced by ☝️

**Pull request closed without merging**